frontend/src/concepts/pipelines/content/StartingStatusModal.tsx

```diff
@@ -0,0 +1,193 @@
+import * as React from 'react';
+import {
+  Modal,
+  ModalBody,
+  ModalHeader,
+  ModalFooter,
+  Spinner,
+  Stack,
+  StackItem,
+  Tab,
+  Tabs,
+  TabTitleText,
+  Panel,
+  PanelMain,
+  Button,
+  Flex,
+  FlexItem,
+  Content,
+  Alert,
+} from '@patternfly/react-core';
+import { usePipelinesAPI } from '#~/concepts/pipelines/context';
+import { getStatusFromCondition } from '#~/concepts/pipelines/content/utils.tsx';
+import K8sStatusIcon, { StatusType } from '#~/concepts/pipelines/content/K8sStatusIcon.tsx';
+import { K8sCondition } from '#~/k8sTypes';
+import {
+  useWatchPodsForPipelineServerEvents,
+  useWatchMultiplePodEvents,
+} from '#~/concepts/pipelines/context/usePipelineEvents.ts';
+import EventLog from '#~/concepts/k8s/EventLog/EventLog';
+import '#~/concepts/dashboard/ModalStyles.scss';
+
+const PROGRESS_TAB = 'Progress';
+const EVENT_LOG_TAB = 'Events log';
+
+type StartingStatusModalProps = {
+  onClose: () => void;
+};
+
+const StartingStatusModal: React.FC<StartingStatusModalProps> = ({ onClose }) => {
+  const { pipelinesServer, namespace } = usePipelinesAPI();
+  const [activeTab, setActiveTab] = React.useState<string>(PROGRESS_TAB);
+  const isServerReadyAndCompletelyDone = pipelinesServer.crStatus?.conditions?.some(
+    (c) => c.type === 'Ready' && c.status === 'True',
+  );
+
+  const [pods] = useWatchPodsForPipelineServerEvents(namespace);
+  const podUids = pods.map((onePod) => onePod.metadata.uid);
+
+  // Use the custom hook to get all events
+  const allEvents = useWatchMultiplePodEvents(
+    namespace,
+    podUids.filter((uid): uid is string => uid !== undefined),
```

Comment:
it might make sense to combine this map and filter into one podUId and put it behind useMemo. Otherwise useWatchMultiplePodEvents will have to rerun each time.